commit e4b8a467514dc29d0021c3fc346340a8037ada8c (refs/changes/66/1466/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-02-07T14:36:34-08:00 (2 years, 8 months ago)
    
    joyent/node-cueball#88 Uncaught Error: Invalid FSM transition: backoff => closed

diff --git a/lib/connection-fsm.js b/lib/connection-fsm.js
index 5723d21..e78abf2 100644
--- a/lib/connection-fsm.js
+++ b/lib/connection-fsm.js
@@ -336,7 +336,7 @@ SocketMgrFSM.prototype.state_error = function (S) {
 };
 
 SocketMgrFSM.prototype.state_backoff = function (S) {
-	S.validTransitions(['failed', 'connecting']);
+	S.validTransitions(['failed', 'connecting', 'closed']);
 
 	/*
 	 * Unfortunately, "retries" actually means "attempts" in the cueball
